src/Microsoft.OData.Core/ODataServiceCollectionExtensions.cs

```diff
@@ -55,15 +55,15 @@ public static IServiceCollection AddDefaultODataServices(this IServiceCollection
             // Finally, register configurable settings.
             var readerSettings = new ODataMessageReaderSettings(odataVersion);
             configureReaderAction?.Invoke(readerSettings);
-            services.AddScoped(sp => readerSettings);
+            services.AddScoped(sp => readerSettings.Clone());

```

Comment:
question: Would there be an issue if we called configure reader action multiple times e.g. if we used a delegate instead of the cloned instance?

Reply:
Are you referring to this approach:

```c#
services.AddScope(_ => {
    var readerSettings = new ODataMessageReaderSettings(...);
    configureReaderAction?.Invoke(readerSettings);
    return readerSettings;
});
```
If so, I considered it (and mentioned it in the attached issue). The reason I opted for cloning instead of that approach is because we can't predict the cost of the delegate since its user defined, for all we know it could be reading from a file or database. On the other hand, since we control the implementation of the `Clone()` method, we can make better assumptions about it cost and we can optimize if it becomes a hotspot, so I think it's safer to err on the side of calling `Clone()` multiple times rather than calling the opaque user-defined delegate multiple times.